### PR TITLE
refactor: rename xxxComponent props

### PR DESCRIPTION
**Pull request by samouss**
**Summary**

This PR rename some props from `xxxComponent` to `xxx` because they don't really accept a component. They just accept anything that can be render by React (aka: [`node`](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes)). Only `Hits` & `InifiniteHits` accept a real component. The principal changes are on the `SearchBox`, otherwise it's mainly docs changes.

I also moved the props related to the rendering from `connectBreadcrumb` to the `Breadcrumb` component since they are unused in the connector.

**SearchBox**:

- `submitComponent` -> `submit`
- `resetComponent` -> `reset`
- `loadingIndicatorComponent` -> `loadingIndicator`

### CODE REVIEW

File: packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap [hunk 1]
```diff
@@ -284,7 +306,9 @@ exports[`Breadcrumb outputs the default breadcrumb with no refinement 1`] = `
     >
       <span
         className="ais-Breadcrumb-separator"
-      />
+      >
+        &gt;
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap]: is this change on purpose? I don't see the change in the test

Comment (samouss) [reply packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap]: Yes since I move the `defaultProps` from the connector to the component it print now the default separator in the snapshot.

Comment (Haroenv) [reply packages/react-instantsearch/src/components/__snapshots__/Breadcrumb.test.js.snap]: Ah okay makes sense

File: packages/react-instantsearch/src/components/Breadcrumb.js [hunk 2]
```diff
@@ -21,12 +21,14 @@ class Breadcrumb extends Component {
...
     rootURL: PropTypes.string,
-    separator: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    separator: PropTypes.node,
     translate: PropTypes.func.isRequired,
     className: PropTypes.string,
   };
 
   static defaultProps = {
+    rootURL: null,
+    separator: '>',
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/Breadcrumb.js]: Should it have spaces around?

Comment (samouss) [reply packages/react-instantsearch/src/components/Breadcrumb.js]: Don't think so, the default theme adds margin around the separator.

Comment (Haroenv) [reply packages/react-instantsearch/src/components/Breadcrumb.js]: Since it's `span` it will get read as `level>sublevel` and might not be what we want

Comment (samouss) [reply packages/react-instantsearch/src/components/Breadcrumb.js]: For screen reader you mean?

Comment (Haroenv) [reply packages/react-instantsearch/src/components/Breadcrumb.js]: yes
